pkg/operatorsource/configuring.go

```diff
@@ -115,38 +113,24 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	packages := r.datastore.GetPackageIDsByOperatorSource(out.GetUID())
 	out.Status.Packages = packages
 
-	cscCreate := new(builders.CatalogSourceConfigBuilder).WithTypeMeta().
-		WithNamespacedName(in.Namespace, in.Name).
-		WithLabels(in.GetLabels()).
-		WithSpec(in.Namespace, packages, in.Spec.DisplayName, in.Spec.Publisher).
-		WithOwnerLabel(in).
-		CatalogSourceConfig()
+	grpcCatalogEnsurer := grpccatalog.NewEnsurer(r.logger, r.reader, r.client)
 
-	err = r.client.Create(ctx, cscCreate)
-	if err != nil && !k8s_errors.IsAlreadyExists(err) {
-		r.logger.Errorf("Unexpected error while creating CatalogSourceConfig: %s", err.Error())
-		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
-		return
+	key := client.ObjectKey{
+		Name:      in.Name,
+		Namespace: in.Namespace,
 	}
-
-	if err == nil {
-		nextPhase = phase.GetNext(phase.Succeeded)
-		r.logger.Info("CatalogSourceConfig object has been created successfully")
-
-		return
+	// Get all labels from OperatorSource and add the DatastoreLabel to the map
+	labels := map[string]string{
+		datastore.DatastoreLabel: "true"}
+	for key, value := range in.Labels {
+		labels[key] = value
 	}
-
-	// If we are here, the given CatalogSourceConfig object already exists.
-	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
+	err = grpcCatalogEnsurer.EnsureResources(key, in.Spec.DisplayName, in.Spec.Publisher, in.Namespace, packages, labels)
```

Comment:
This call should indicate using a `bool` that you are creating a datastore `CatalogSource` and not pass in labels.

Reply:
From https://github.com/operator-framework/operator-marketplace/pull/198#discussion_r290855167 since we need to pass in labels, we can just append to it the datastore label that needs to be propagated.